tests/test_predefined_driven_controls.py

```diff
@@ -550,6 +551,33 @@ def test_walsh_control():
     assert np.allclose(pi_on_4_segments, _pi_on_4_segments)
 
 
+def test_gaussian_control():
+    """
+    Tests Gaussian control.
+    """
+    _rabi_rotation = np.pi
+    _segment_count = 10
+    _duration = 0.01
+    _width = 0.002
+
+    gaussian_control = new_gaussian_control(
+        rabi_rotation=_rabi_rotation,
+        segment_count=_segment_count,
+        duration=_duration,
+        width=_width,
+    )
+
+    # compute total rotation of generated pulse
+    rabi_rotation = np.dot(gaussian_control.rabi_rates, gaussian_control.durations)
+
+    # check number and duration of pulses
+    assert len(gaussian_control.rabi_rates) == _segment_count
+    assert np.allclose(gaussian_control.durations, _duration / _segment_count)
+
+    # check total rotation of pulse
+    assert np.isclose(rabi_rotation, _rabi_rotation)
```

Comment:
What if we also tested that the shape of controls matches the expected Gaussian, just like the next test checks that the `segment_values` are all close to the `expected_gaussian`?

Reply:
Thank you! Unfortunately, I think one has to discretize the Gaussian first anyways to determine the proper normalization factor, so it ends up looking quite similar to the implementation itself. But would something like this work?
```suggestion
    # determine the segment mid-points
    segment_mid_points = _duration / _segment_count * (0.5 + np.arange(_segment_count))
    base_gaussian_mean = _duration * 0.5
    base_gaussian_width = _width
    base_gaussian = np.exp(
        -0.5 * ((segment_mid_points - base_gaussian_mean) / base_gaussian_width) ** 2.0
    )

    # translate pulse by B/A (from Motzoi paper) to ensure output is 0 at t=0
    y_translation = -np.exp(-0.5 * ((0 - base_gaussian_mean) / base_gaussian_width) ** 2)
    base_gaussian += y_translation
    base_gaussian_total_rotation = np.sum(base_gaussian) * _duration / _segment_count
    expected_gaussian = (
        base_gaussian / base_gaussian_total_rotation
    ) * _rabi_rotation

    # check number and duration of pulses
    assert len(gaussian_control.rabi_rates) == _segment_count
    assert np.allclose(gaussian_control.durations, _duration / _segment_count)

    # check total rotation of pulse
    assert np.isclose(rabi_rotation, _rabi_rotation)
    
    # check shape of controls matches the expected Gaussian
    assert np.allclose(gaussian_control.rabi_rates, expected_gaussian)
```

Reply:
That's a good point, reproducing the logic of the function in the test is probably not ideal. As @tachikoma-li mentioned below, we can probably replace A and B with the expected values in this example. I'd go further and say we probably can eliminate A from the check that is confirming that the shape is Gaussian, as the value of A is already being checked by `assert np.isclose(rabi_rotation, _rabi_rotation)`. In this case, maybe we can use something like the following as the extra check:
```
times = np.linspace(0.0005, 0.0095, 10)
gauss = lambda t: np.exp(-0.5*((t-_duration/2)/_width)**2)

expected_normalized_pulse = (gauss(times) - gauss(0))/max(gauss(times) - gauss(0))
normalized_pulse = gaussian_control.rabi_rates/max(gaussian_control.rabi_rates)

assert np.allclose(expected_normalized_pulse, normalized_pulse)
```

Reply:
Good idea, decoupling the shape check from the normalization check would be better.

```suggestion
    _segment_width = _duration / _segment_count
    midpoints = np.linspace(
        _segment_width / 2, _duration - _segment_width / 2, _segment_count
    )

    def gauss(t):
        return np.exp(-0.5 * ((t - _duration / 2) / _width) ** 2)

    expected_normalized_pulse = (gauss(midpoints) - gauss(0)) / max(
        gauss(midpoints) - gauss(0)
    )
    normalized_pulse = gaussian_control.rabi_rates / max(gaussian_control.rabi_rates)

    # check pulse is Gaussian-shaped
    assert np.allclose(expected_normalized_pulse, normalized_pulse)

    # compute total rotation of generated pulse
    rabi_rotation = np.dot(gaussian_control.rabi_rates, gaussian_control.durations)

    # check number and duration of pulses
    assert len(gaussian_control.rabi_rates) == _segment_count
    assert np.allclose(gaussian_control.durations, _duration / _segment_count)

    # check total rotation of pulse
    assert np.isclose(rabi_rotation, _rabi_rotation)
```

Reply:
Cool, that looks good to me. @tachikoma-li Any further thoughts about the test?

Reply:
LGTM. Nice.